manifests/install.pp

```diff
@@ -27,10 +27,25 @@
       $provider = undef
     }
 
-    package { $::jmxtrans::package_name:
-      ensure   => $::jmxtrans::package_version,
-      provider => $provider,
-      source   => $::jmxtrans::package_source,
+    # dpkg does'nt support installing from a URL, so this downloads a source file from a URL if one is provided.
+    if $provider == 'dpkg' and $::jmxtrans::package_source.match(/^http.*/) {
+
+      archive { '/tmp/jmxtrans.deb':
+        source => $::jmxtrans::package_source,
+      }
+
+      package { $::jmxtrans::package_name:
+        ensure   => $::jmxtrans::package_version,
+        provider => $provider,
+        source   => '/tmp/jmxtrans.deb',
+      }
```

Comment:
Won't this need `require => Archive['/tmp/jmxtrans.deb'],`?

Reply:
You're right!  I fixed this and cleaned up the top-scope variables